ClientCore/Enums/Rank.cs

```diff
@@ -0,0 +1,11 @@
+﻿
+namespace ClientCore.Enums
+{
+    public enum Rank
```

Comment:
```suggestion
    public enum Rank : int
```

and remove unnecessary type casts